### ISSUE TITLE
Bug when creating rational intervals

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/ValidatedNumerics.jl#94]
```
Interval{Rational{Int}}(big(2))
```
gives an error with the current definition of `make_interval` for intervals of rationals.

I have to say that I am inclined to remove the rational intervals completely... cc @lbenet

**Comment by lbenet** [comment]
I don't think this is a bug, simply, that some promotion is not working, since `Interval{Rational{BigInt}}(big(2))` gives te expected result.

I think rational intervals should be there since they are exact.

**Comment by dpsanders** [comment]
Closing since the following works:
`convert(Interval{Rational{Int}}, big(2))`

**Comment by lbenet** [comment]
So, why is `Interval{Rational{Int}}(big(2))` not working?

**Comment by lbenet** [comment]
The changes implemented in #98 still give the same error:
```julia
julia> Interval{Rational{Int}}(big(2))
ERROR: TypeError: call: in typeassert, expected ValidatedNumerics.Interval{Rational{Int64}}, got ValidatedNumerics.Interval{Rational{BigInt}}
 in call at ./essentials.jl:56
```
I'm reopening this issue to keep it in mind.

**Comment by dpsanders** [comment]
`Interval{Rational{Int}}(big(2))` does not work since this just tries to call the constructor of `Interval{Rational{Int}}`, which does no conversion and requires to be passed a `Rational{Int}`, not a `BigInt`.
